PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -832,4 +842,42 @@ bool str2Clipboard(const generic_string &str2cpy, HWND hwnd)
 		return false;
 	}
 	return true;
-}
\ No newline at end of file
+}
+
+// string trimming; algorithms taken from http://stackoverflow.com/questions/216823/whats-the-best-way-to-trim-stdstring
+generic_string stringTrimLeft(const generic_string & s)
+{
+	generic_string result(s);
+	result.erase(result.begin(), std::find_if(result.begin(), result.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
+	return result;
+}
+
+generic_string stringTrimRight(const generic_string & s)
+{
+	generic_string result(s);
+	result.erase(std::find_if(result.rbegin(), result.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), result.end());
+	return result;
+}
+
+generic_string & stringTrimLeftInline(generic_string & result)
+{
+	result.erase(result.begin(), std::find_if(result.begin(), result.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
+	return result;
+}
+
+generic_string & stringTrimRightInline(generic_string & s)
+{
+	s.erase(std::find_if(s.rbegin(), s.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
+	return s;
+}
+
+generic_string stringTrim(const generic_string &s)
+{
+	generic_string result(s);
+	return stringTrimLeftInline(stringTrimRightInline(result));

```

Comment:
Here, the `StringTrimLeftInline` returned reference is being copy constructed to a value..... why not just:


```
	generic_string result(s);
	stringTrimRightInline(result)
	stringTrimLeftInline(result);
	return result;
```